Update code example to match new sdl2.nim syntax

ldhieu opened the pull request:
I've updated the code example for hot-code reloading to match the new sdl2 syntax at [nim-lang/sdl2](https://github.com/nim-lang/sdl2/blob/master/src/sdl2.nim)

I've also added a reminder for people who encounter the error `dlopen(libnimhcr.dylib, 2): image not found` as I did earlier today.

Please review and tell me if anything is amiss. This is my first time doing this.

Cheers,
Hieu

ldhieu committed:
Update code example to match new sdl2.nim syntax

Signed-off-by: Hieu Luu Danh <hieu@vivu.asia>

alaviss commented on doc/hcr.rst:
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
-  var dY = 1
-
+  
+  var posX : cint = 1
+  var posY : cint = 0
+  var dX : cint = 1
+  var dY : cint = 1
```
```suggestion
  var posX: cint = 1
  var posY: cint = 0
  var dX: cint = 1
  var dY: cint = 1
```

Some minor cosmetic changes.

ldhieu replied:
commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this

alaviss commented on doc/hcr.rst:
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
-        if evt.key.keysym.scancode == SCANCODE_ESCAPE: runGame = false
-        elif evt.key.keysym.scancode == SCANCODE_F9:
+      if evt.kind == KeyDown:
+        if evt.key.keysym.scancode == getScancodeFromName("Escape"): runGame = false
+        elif evt.key.keysym.scancode == getScancodeFromName("F9"):
```
Looks like you can still use the constants, they have `SDL_` prefixes: https://github.com/nim-lang/sdl2/blob/46aa2c0e30e9edf6028bc432a3ed70e51a0fe9f9/src/sdl2/private/keycodes.nim#L4

ldhieu replied:
Thanks! I see them now. Commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this.

alaviss commented on doc/hcr.rst:
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
-  import sdl2/sdl
-
+  import sdl2
+  
```
Can you cleanup the trailing whitespaces?

ldhieu replied:
I've cleaned up the whitespaces in commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this

alaviss commented on doc/hcr.rst:
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
+    discard renderer.clear()
+    discard renderer.setDrawColor(255, 128, 128, 0)
+  
+    var rect: Rect
+    rect = (x: posX - 25, y: posY - 25, w: 50.cint, h: 50.cint)
```
```suggestion
    var rect: Rect = (x: posX - 25, y: posY - 25, w: 50.cint, h: 50.cint)
```

alaviss commented on doc/hcr.rst:
```diff
@@ -26,59 +26,61 @@ To install SDL2 you can use ``nimble install sdl2``.
+  
+    var rect: Rect
+    rect = (x: posX - 25, y: posY - 25, w: 50.cint, h: 50.cint)
+    discard renderer.fillRect(rect.addr)
```
```suggestion
    discard renderer.fillRect(rect)
```

There's a variant that takes a `var Rect`.

ldhieu replied:
Cheers alaviss
commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` resolved this

alaviss commented on doc/hcr.rst:
```diff
@@ -187,7 +189,9 @@ Native code targets
-hot code reloading run-time.
+hot code reloading run-time. Both libraries can be found in the ``lib``
+folder of Nim and can be compiled to a ``.dylib`` file to satisfy compiling
+of the example code above.
```
I think this should describe how to compile them instead.

ldhieu replied:
I've added some example console commands on UNIX/MacOS

I was not sure whether to use RST `::` or use `.. code-block:: console` (used the former as it's how it's written in this document)

commit `daccd8464b4186013ef5b59d13b3939512e2fdc1` is a suggestion on this.

ldhieu committed:
Modify on recommendation of @alaviss

- Removed trailing whitespaces
- Detailed how to compile libnimhcr and libnimrtl
- Fixed some logic in example code

ldhieu commented:
Hello, thank you for your comments @alaviss ! I've added the changes and noted you in the commit message.

Let me know if something else is needed !

Hieu

alaviss commented on doc/hcr.rst:
```diff
@@ -187,7 +188,25 @@ Native code targets
 the ``nimrtl`` library and the ``nimhcr`` library which implements the
-hot code reloading run-time.
+hot code reloading run-time. Both libraries can be found in the ``lib``
+folder of Nim and can be compiled to a ``.dylib`` file to satisfy compiling
```
Please note that `lib<name>.dylib` is the format for macOS. This differs between operating systems (`<name>.dll` on Windows and `lib<name>.so` on *nix).

I think we can just get away with "compiled into dynamic libraries".

> satisfy compiling of the example code above.

It's actually only needed at runtime.

ldhieu replied:
Thank you ! Absolutely right.

I've committed `6b31eb153e29ed670141689f71e70d8ef5aeba79` to resolve this.

ldhieu committed:
Modify following recommendations of @alaviss

- Rewording so that it conforms to Windows/Linux/MacOS

Signed-off-by: Hieu Luu Danh <hieu@vivu.asia>

Araq commented:
Note: I need to test this manually before merging.
